### PR TITLE
Milestone-3: Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by MNisarAli**
### ♻️ Changes Made:

- [x] Added validations to the `User` and `Post` models to ensure data integrity.
- [x] Created unit specs for all 'models', 'methods', and 'validations'.

### 👀 Impact:

- The project now has validations to prevent data inconsistencies and ensure data quality
- The unit specs ensure that all models' methods and validations are working as expected
- The project is now ready for further development and testing

### 🙏 Thank you for considering this pull request! 🙏

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -54,6 +54,7 @@ gem 'bootsnap', require: false
 group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
   gem 'debug', platforms: %i[mri mingw x64_mingw]
+  gem 'rspec-rails'
```
Comment (mukhammadrizooff) [inline Gemfile:57]: - [ ] [REQUIRED] Good job so far! When I check `"model specs"` they all fail, Please fix the errors in your `specs` 🙏🏼 

image

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @MNisarAli, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile (hunk 1 above)
Comment (MNisarAli) [reply Gemfile:57]: Thank you for your thorough and insightful review of my project. I have fixed 💯 the errors in the `model specs`. 👍

**Review by mikethreels — APPROVED** [review]
Hi Nisar, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job making the requested changes
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**